tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,257 @@
+        organization=module_sca_manifest_org
+    ).create()
+    c_view = module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+    return {
+        'organization': module_sca_manifest_org,
+        'satellite': module_target_sat,
+        'environment': lc_env,
+        'contentview': c_view,
+    }
+
+
+@pytest.mark.no_containers
+@pytest.mark.parametrize(
+    'custom_host',
+    [
+        {'deploy_rhel_version': '8.8'},
+    ],
+    ids=['RHEL8.8'],
+    indirect=True,
+)
+def test_upgrade_rhel8_to_rhel9(
+    setup_env,
+    custom_host,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 7add9b32-a13a-4e65-973c-cd64326b332a
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create LCE, Create CV, add repositories to it, publish and promote CV, Create AK, etc.
+        3. Register content host with AK
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    # rhel_contenthost = custom_host
+    organization = setup_env['organization']
+    target_sat = setup_env['satellite']
+    lc_env = setup_env['environment']
+    cv = setup_env['contentview']
+
+    # Enable rhel8/rhel9 bos, aps repository and add in content view
+    all_repos = []
+    for rh_repo_key in CUSTOM_REPOS.keys():
+        prod = rh_repo_key.split('_')[0]
+        repo_id = target_sat.api_factory.enable_rhrepo_and_fetchid(
+            basearch=custom_host.arch,
+            org_id=organization.id,
+            product=PRDS[prod],
+            repo=CUSTOM_REPOS[rh_repo_key]['name'],
+            reposet=CUSTOM_REPOSET[rh_repo_key],
+            releasever=CUSTOM_REPOS[rh_repo_key]['releasever'],
+        )
+        rh_repo = target_sat.api.Repository(id=repo_id).read()
+        all_repos.append(rh_repo)
+        # sync repo
+        rh_repo.sync(timeout=1800)
+    cv.repository = all_repos
+    cv = cv.update(['repository'])
+    # Publish, promote content view to lce
+    cv.publish()
+    cvv = cv.read().version[0]
+    cvv.promote(data={'environment_ids': lc_env.id, 'force': True})
+    cv = cv.read()
+    # Create activation key
+    ak = create_activation_key(target_sat, cv, lc_env, organization)
+
+    # 3. Register Host
+    register_host_with_satellite(target_sat, custom_host, organization, ak)
+
+    # 4. Verify target rhel version repositories has enabled on Satellite Server
+    verify_target_repo_on_satellite(target_sat, cv, organization, lc_env, target_rhel='rhel-9')
+
+    # 5. Update all packages and install Leapp utility
+    # Preupgrade conditions and check
+    rhel_old_ver = custom_host.run('cat /etc/redhat-release')
```

Comment:
Please use `custom_host.os_version` to fetch the system version.
With #12144 you can clean the cached properties and use `os_version` to fetch the OS version after the upgrade.